### ISSUE TITLE
Receiving HM02 Gives TM00 (Item Overflow)

### ISSUE THREAD

**Issue by Matando** [roukaour/polishedcrystal#32]
Upon receiving HM02 Fly I also received TM00 Tackle.

It appears to be an item overflow bug within the TM/HM pocket as having 16 TM/HMs causes the issue.

Also Strength becomes the cancel button and Cancel is now an HM for aqua tail.

git rev-parse HEAD reports : d36bc3de4df94506b03295348b56e89d61e2a457

I know this isn't the latest version but trying to use an old save with a new revision doesn't load.

I can provide a sav file upon request.

**Comment by roukaour** [comment]
A save file would be helpful. I cannot reproduce this.

**Comment by Matando** [comment]
Here is a copy of my save file. It's happened to me on more than one playthrough although this is the only save file I still have.  Last time it happened when I got Strength.

[polishedcrystal-faithful-revd36bc3d.sav.zip](https://github.com/roukaour/polishedcrystal/files/1041783/polishedcrystal-faithful-revd36bc3d.sav.zip)

**Comment by roukaour** [comment]
Thank you.

Whatever is causing this bug is unpredictable. I started a new game with a cheat script to give your 15 TMs+HMs, and then getting HM02 Fly from Walker worked normally. Do you happen to have a save from just before the battle?

**Comment by Matando** [comment]
Unfortunately I don't. But since this has happened to me a couple of times. I did purchase a bunch of healing items previously as I was having trouble beating Walker.  I also did try continuing on with a previous save and it is game breaking because I wasn't able to teach anyone waterfall due to the bug.

I will do another play through to see if I can more reliably produce the bug with a save file from before and after.
